src/layouts/Sites.tsx

```diff
@@ -63,41 +113,30 @@ const SitesContent = ({ siteNames }: { siteNames?: SiteData[] }) => {
         </VStack>
       </VStack>
     )
+  const { data: previews } = useSitePreview(
+    email,
+    siteNames.map((site) => site.repoName)
+  )
+  if (previews && previews.length === siteNames.length) {
```

Comment:
Hmm sorry I dont understand this code :(
we only render this when we get the preview images for __**all**__ sites have been loaded? shouldnt we load it eagerly, ie for each site that has a preview, use the url for that site?

Reply:
The if statement `if (previews && previews.length === siteNames.length) {` only serves as a out of bounds check as the if frontend sends an array of sites to request for previews, it should expect an array of the same length back to frontend. We chose to use a list of sites as 1 request because for admin users that has access to 1k repos, it was making 1k requests to the backend which completely flood the all logs and gets ratelimited after the first 100 or so.

Reply:
> it was making 1k requests to the backend which completely flood the all logs and gets ratelimited after the first 100 or so

Hmm why would we get rate limited ah?

Reply:
ya shouldnt be 1k calls bah regardless ah

Reply:
there's a ratelimiter in `src/services/utilServices/RateLimiter.ts` which limits to around 200 calls per 15mins. Admins completely exhaust this and can no longer perform any actions.